drivers/spi/spi.c

```diff
@@ -85,3 +85,34 @@ int32_t spi_write_and_read(struct spi_desc *desc,
 {
 	return desc->platform_ops->write_and_read(desc, data, bytes_number);
 }
+
+/**
+ * @brief  Iterate over head list and send all spi messages
+ * @param desc - The SPI descriptor.
+ * @param head - First link of the list with messages to be sent
+ * @return SUCCESS in case of success, negativ error code otherwise.
+ */
+int32_t spi_write_and_read_list(struct spi_desc *desc,
```

Comment:
I think `spi_write_and_read` is named like this for historical reasons.
In SPI there is always write and read at the same time on MOSI and MISO respectively.
A better name would have been `spi_transfer` (2 words vs 4 words).
In this case, since it's a new function and we can name it whatever, we could use `spi_multi_transfer` (3 words vs 5 words)

Reply:
renamed to spi_transfer